CMakeLists.txt

```diff
@@ -1,3 +1,27 @@
+# Check ESP-IDF version and error out if it is not in the supported range.
+#
+# Note for arduino-esp32 developers: to bypass the version check locally,
+# set ARDUINO_SKIP_IDF_VERSION_CHECK environment variable to 1. For example:
+#   export ARDUINO_SKIP_IDF_VERSION_CHECK=1
+#   idf.py build
+
+set(min_supported_idf_version "4.4.0")
+set(max_supported_idf_version "4.4.99")
+set(idf_version "${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}.${IDF_VERSION_PATCH}")
+
```

Comment:
It may be simpler to test the IDF version as:
```
if (NOT IDF_VERSION_MAJOR EQUAL 4 AND NOT IDF_VERSION_MINOR EQUAL 4)
  message(FATAL_ERROR "Arduino-esp32 can be used with ESP-IDF version 4.4.x"
                        "but an older version is detected: ${idf_version}.")
endif()
```

Reply:
I agree, I just wanted to write it in a way such that later Arduino-esp32 developers only need to modify version numbers and don't have to deal with CMake if statement syntax, in case more than one minor version gets supported.